### PR TITLE
Issue #89: agent for tracking unaligned memory accesses with buffers

### PR DESCRIPTION
**Pull request by m-anthony**
as discussed in issue #89, check for unaligned accesses can be useful during development/testing but modifying the code is not an option since it can impact inlining. This agent should allow this check when required.

I modified the build.gradle file to generate a new artifact (with classifier 'agent') that contains only the agent classes. I wanted to use the shadow plugin to include bytebuddy  in the agent jar, however I didn't succeed to include it without including all other agrona classes or creating a multi-projects build

Any help or hint to improve the packaging is welcome

### CODE REVIEW

File: src/test/java/org/agrona/agent/BufferAlignmentAgentTest.java [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    @Test
+    public void testUnsafeBufferFromByteArray()
+    {
+        testUnsafeBuffer(new UnsafeBuffer(new byte[256]), 0);
+    }
+
+    @Test
+    public void testUnsafeBufferFromByteArrayWithOffset()
+    {
+        final UnsafeBuffer buffer = new UnsafeBuffer(new byte[256], 1, 128);
```
Comment (mjpt777) [inline src/test/java/org/agrona/agent/BufferAlignmentAgentTest.java]: On 32bit JVMs this tends to fail. If you use ByteBuffer.allocateDirect() then is will be 8 byte aligned.

**Comment by mjpt777** [comment]
If a new module is created for the agent then the packaging should be cleaner. If you look at Aeron you can see how a module can be used.

File: src/test/java/org/agrona/agent/BufferAlignmentAgentTest.java (hunk 1 above)
Comment (m-anthony) [inline src/test/java/org/agrona/agent/BufferAlignmentAgentTest.java]: Good point, I totally missed than Unsafe.ARRAY_BYTE_BASE_OFFSET is not necessary a multiple of 8. I definitely need to update the heap-based tests to handle this.

However, do you think that all this alignment stuff deserve a better documentation ? Unaligned accesses can be really nasty on some platforms, and such subtle things can easily be missed when trying get correct alignment

Another related concern is the behavior of ExpandableArrayBuffer when Unsafe.ARRAY_BYTE_BASE_OFFSET is not a multiple of 8. Does it make sense to add another offset to make sure that the address of index 0 in the buffer is a multiple of 8 ?

Let me know if you want to create side issues for these concerns.

**Comment by m-anthony** [comment]
I wanted to avoid switching to a multi-module build just for a debug feature, but if there is no other clean way to package it correctly, I will do the change.

**Comment by m-anthony** [comment]
@mjpt777 I did some fixes on this PR, can you please have another look at it ? Thanks.

**Comment by mjpt777** [comment]
Seems reasonable if you can resolve current conflicts.

**Comment by m-anthony** [comment]
@mjpt777 conflicts are now resolved

**Comment by mjpt777** [comment]
Thanks

**Comment by mjpt777** [comment]
Could you add a Wiki page on how to use it?

**Comment by m-anthony** [comment]
I added a wiki page.
Feel free to review/edit it. Since I'm a non-native english speaker, it can certainly be improved.

**Pull request merged**